apps/api/src/app/workflows-v2/usecases/overload-content-data/overload-content-data-on-workflow.usecase.ts

```diff
@@ -89,11 +88,15 @@ export class OverloadContentDataOnWorkflowUseCase {
     workflow: WorkflowInternalResponseDto,
     stepIdToControlValuesMap: { [p: string]: ValidatedContentResponse }
   ) {
+    if (workflow.origin === WorkflowOriginEnum.EXTERNAL) {
+      return;
+    }
```

Comment:
was needed because this flow was overriding the payload schema on the upsert request.

Reply:
This feels like a code-smell. We shouldn't treat workflows differently when generating the payload sample. Please elaborate on the underlying issue and let's work it out differently.

Reply:
If the workflow doesn't have a payload schema, the use case that generates the sample payload, should use the presence or absence of the workflow.payload schema in the database to handle the extra.

Again the argument is the same, use a single flow and use case for that

Reply:
this comment resolves https://github.com/novuhq/novu/pull/7137/files#r1861939413 as well.

Reply:
This was resolved offline.